### ISSUE TITLE
idx has odd size

### ISSUE THREAD

**Issue by tijmengit** [lindsey98/Phishpedia#2]
I am trying to run the phishpedia_main.py on the alibaba example. Though, my runs keep failing due to lack of memory. When inspecting the program, in siamese_pedia/inference.py, at line 105, idx is set. If I understand correctly, this should hold the indexes of the top 10 brands. Though, in my run, this variable has a shape of 10x2048x2048. Am I doing something wrong here? What shape is expected here?

**Comment by lindsey98** [comment]
Hi,
   Thanks for raising the issue, I believe there is a shape mismatch bug in my code. I have fixed the bug in my latest commit, please pull and have a try again. As for the shape of idx, yes you are right it should hold the indices, it should be a single-dimensional list of shape (topK,). 
   In addition I replace a better test site (a phishing one), it is expected to return "Google" as targeted brand. If it does not, please let me know again.

**Comment by lindsey98** [comment]
Hi, 
   Can you please check src/siamese_pedia/siamese_retrain/bit_pytorch/models.py line 155, does it look like this? 
![Screenshot 2021-09-02 at 8 35 17 PM](https://user-images.githubusercontent.com/51521323/131844734-4ef0b26e-1ae8-4097-a21e-6b898e009dca.png)

**Comment by lindsey98** [comment]
Can I double check have you pulled my latest commits on main branch? I believe I have already corrected those files.

**Comment by tijmengit** [comment]
I did indeed pull the latest commits. I am not sure what is going wrong here. Though, I did create a working version for myself. 
I calculate the cosine similarity not by taking the dot product, but use the sklearn.metric.pairwise.cosine_similarity. 
The sim list is filled by:
```
    sim_list = []
    for logo in logo_feat_list:
        similarity = cosine_similarity(logo.T, img_feat.T)
        sim_list.append(similarity[0][0])
```
This should result in a similar result right? I have been testing this on the dataset, and results are good.
